context.go

```diff
@@ -19,58 +19,34 @@ import (
 
 // Ctx defines the supporting context of the tool.
 type Ctx struct {
+	WorkingDir string // Where to execute
+	Out, Err   *log.Logger
+	Verbose    bool     // Enables more verbose logging
 	GOPATH     string   // Selected Go path
 	GOPATHS    []string // Other Go paths
-	WorkingDir string
-	*Loggers
 }
 
-// Loggers holds standard loggers and a verbosity flag.
-type Loggers struct {
-	Out, Err *log.Logger
-	// Whether verbose logging is enabled.
-	Verbose bool
-}
-
-// NewContext creates a struct with the project's GOPATH. It assumes
+// SetGOPATH sets the project's GOPATH from an environment var. It assumes
 // that of your "GOPATH"'s we want the one we are currently in.
-func NewContext(wd string, env []string, loggers *Loggers) (*Ctx, error) {
-	ctx := &Ctx{WorkingDir: wd, Loggers: loggers}
-
-	GOPATH := getEnv(env, "GOPATH")
-	if GOPATH == "" {
-		GOPATH = defaultGOPATH()
+func (c *Ctx) SetGOPATH(gopath string) error {
```

Comment:
Mmm, ok - given that my objection is about a suggestion to the user, I think that as long as the documentation clearly cautions the user appropriately, then it's OK.